spec/support/cancan_helper.rb

```diff
@@ -13,15 +13,18 @@ module Spree
       ability_hash.each do |action, _true_or_false|
         @ability_result[action] = ability.can?(action, target)
       end
-      ability_hash.diff(@ability_result).none?
+
+      ability_hash.inject(true) do |acc, (k, v)|
+        acc && @ability_result[k] == v
+      end
```

Comment:
I fixed this in the rails 4.1 recently.
I ended up with this (this function comes from minitest that comes with activesupport, maybe we can add it as an explicit dependency in our gemfile):`
assert_equal(ability_hash, @ability_result)`
I think this is easier than your solution. I wonder if this works correctly and still removes the warning. Could you give it a try?

Reply:
@luisramos0 I ended up using RSpec's equality matcher since we heavily use RSpec in the project.

Reply:
I thought that would not work! Nice, the build is green :+1: